spec/rubocop/cop/rspec/dialect_spec.rb

```diff
@@ -1,59 +1,186 @@
 # frozen_string_literal: true
 
 RSpec.describe RuboCop::Cop::RSpec::Dialect do
-  let(:cop_config) do
-    {
-      'PreferredMethods' => {
-        'context' => 'describe'
-      }
-    }
-  end
+  context 'with preferred methods' do
+    context 'when `describe` is preferred to `context`' do
+      let(:cop_config) do
+        {
+          'PreferredMethods' => {
+            'context' => 'describe'
+          }
+        }
+      end
 
-  it 'allows describe blocks' do
-    expect_no_offenses(<<~RUBY)
-      describe 'display name presence' do
+      it 'allows describe blocks' do
+        expect_no_offenses(<<~RUBY)
+          RSpec.describe 'context' do
+            describe 'display name presence' do
+            end
+          end
+        RUBY
       end
-    RUBY
-  end
 
-  it 'allows calling methods named context in examples' do
-    expect_no_offenses(<<~RUBY)
-      it 'tests common context invocations' do
-        expect(request.context).to be_empty?
+      it 'registers an offense for context blocks' do
+        expect_offense(<<~RUBY)
+          RSpec.describe 'context' do
+            context 'display name presence' do
+            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer `describe` over `context`.
+            end
+          end
+        RUBY
+
+        expect_correction(<<~RUBY)
+          RSpec.describe 'context' do
+            describe 'display name presence' do
+            end
+          end
+        RUBY
       end
-    RUBY
-  end
+    end
 
-  it 'registers an offense for context blocks' do
-    expect_offense(<<~RUBY)
-      context 'display name presence' do
-      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer `describe` over `context`.
+    context 'when `describe` is preferred to `feature`' do
+      let(:cop_config) do
+        {
+          'PreferredMethods' => {
+            'feature' => 'describe'
+          }
+        }
       end
-    RUBY
 
-    expect_correction(<<~RUBY)
-      describe 'display name presence' do
+      it 'allows describe blocks' do
+        expect_no_offenses(<<~RUBY)
+          RSpec.describe 'context' do
+            describe 'display name presence' do
+            end
+          end
+        RUBY
       end
-    RUBY
-  end
 
-  it 'registers an offense for RSpec.context blocks' do
-    expect_offense(<<~RUBY)
-      RSpec.context 'context' do
-      ^^^^^^^^^^^^^^^^^^^^^^^ Prefer `describe` over `context`.
-        it 'tests common context invocations' do
-          expect(request.context).to be_empty?
```

Comment:
Those ones were here to ensure that `context` with an explicit receiver is not detected. Can you keep it in some way?

Reply:
Sorry for the late reply 🙏 

> Those ones were here to ensure that `context` with an explicit receiver is not detected. Can you keep it in some way?

In fa5da43 the check using `rspec_method?` has been removed, so explicit receivers are also detected. To avoid detection, I can't think of any ideas other than to do the following.

```diff
          def on_send(node)
            return unless inside_example_group?(node)
+           return unless rspec?(node.children.first)
            return unless preferred_methods[node.method_name]
```

Do you have any good ideas?